pkg/transit/transit.go

```diff
@@ -0,0 +1,161 @@
+package transit
+
+import (
+	"encoding/base64"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/hashicorp/vault/api"
+)
+
+// VaultEncryptor implements Encryptor interface by using Vault transit
+type VaultEncryptor struct {
+	vault   *api.Client
+	appName string
+}
+
+// NewVaultEncryptor instantiate a vault encryption service
+func NewVaultEncryptor(addr, token, appName string) *VaultEncryptor {
+	if addr == "" {
+		addr = "http://localhost:8200"
+	}
+	if token == "" {
+		panic("VAULT_TOKEN is missing")
+	}
+	if appName == "" {
+		panic("VAULT_APP_NAME is missing")
+	}
+
+	config := &api.Config{
+		Address: addr,
+		Timeout: time.Second * 2,
+	}
+
+	client, err := api.NewClient(config)
+	if err != nil {
+		panic(err)
+	}
+	client.SetToken(token)
+
+	s := &VaultEncryptor{
+		appName: appName,
+		vault:   client,
+	}
+
+	ok, err := s.transitKeyExists()
+	if err != nil {
+		panic(err)
+	}
+	if !ok {
+		err = s.transitKeyCreate()
+		if err != nil {
+			panic(err)
+		}
+		log.Println("INFO: Transit key created")
+	}
+
+	err = s.startRenewToken(token)
+	if err != nil {
+		panic(err)
+	}
+
+	return s
+}
+
+func (s *VaultEncryptor) transitKeyExists() (bool, error) {
+	secret, err := s.vault.Logical().Read("transit/keys/" + s.appName)
+	if err != nil {
+		return false, err
+	}
+	return secret != nil, nil
+}
+
+func (s *VaultEncryptor) transitKeyCreate() error {
+	_, err := s.vault.Logical().Write("transit/keys/"+s.appName, map[string]interface{}{
+		"force": true,
+	})
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// Encrypt the plaintext argument and return a ciphertext string or an error
+func (s *VaultEncryptor) Encrypt(plaintext string) (string, error) {
```

Comment:
We should change the interface and the method here to accept an appName and use the respective encryption key